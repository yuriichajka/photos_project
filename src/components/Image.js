import React from 'react';

import { Img } from '../styled/styledPhotos';

const Image = ({ item, onClick }) => <Img loading='lazy'
                                          width='300px'
                                          height='200px'
                                          src={ item }
                                          onClick={ onClick }/>

export default Image;