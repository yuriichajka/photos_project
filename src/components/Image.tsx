import React from 'react';

import { Img } from '../styled/styledPhotos';

interface ImageProps {
    item?: string;
    onClick?: () => void;
}

const Image: React.FC<ImageProps> = ({item, onClick}) => (
    <Img
        className='test'
        loading='lazy'
        width='300px'
        height='200px'
        src={ item }
        onClick={ onClick }
    />
);

export default Image;
