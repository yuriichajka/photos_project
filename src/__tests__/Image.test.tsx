import React from 'react';

import Image from '../components/Image';

import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'

let wrapper;
describe('Image snapshot',  () => {
    it('should add a snapshot', () => {
        wrapper = renderer.create(<Image/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    })
})
