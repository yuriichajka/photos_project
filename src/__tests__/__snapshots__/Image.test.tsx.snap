// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot should add a snapshot 1`] = `
<img
  className="sc-bdnxRM hStcCx test"
  height="200px"
  loading="lazy"
  width="300px"
/>
`;
